examples/binaries/waiter/tests/rw_lock_access.rs

```diff
@@ -0,0 +1,182 @@
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Read, LockStaticAccessSubcommand::Deref),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Read lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn drop_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::Drop),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn as_ref_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::AsRef),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn as_mut_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::AsMut),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn deref_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
+
+    let lock_access_result = program.send(
+        USER_ID,
+        Command::RwLockStaticAccess(RwLockType::Write, LockStaticAccessSubcommand::Deref),
+    );
+
+    assert_paniced(
+        &lock_access_result,
+        &format!(
+            "Write lock guard held by message {} is being accessed by message {}",
+            lock_msg_id,
+            lock_access_result.sent_message_id()
+        ),
+    );
+}
+
+#[test]
+fn deref_mut_w_lock_guard_from_different_msg_fails() {
+    let system = System::new();
+    let (program, lock_msg_id) = init_fixture(&system, RwLockType::Write);
```

Comment:
all the file seems so copy-pasty so I'd introduce common fn for them and reuse for each tests with 2 args: Command and panic message

Reply:
Nice catch. Shame on me I haven't noticed it myself. Addressed